### PR TITLE
Add Grafana Agent to list of distributions

### PR DESCRIPTION
**Pull request by jpkrohling**
This adds Grafana Agent as a distribution and mark the respective components as being present there.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@mattdurham, could you please review this one?

File: cmd/mdatagen/statusdata.go [hunk 1]
```diff
@@ -12,13 +12,14 @@ import (
...
-	"contrib":  "https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib",
-	"aws":      "https://github.com/aws-observability/aws-otel-collector",
-	"observiq": "https://github.com/observIQ/observiq-otel-collector",
-	"splunk":   "https://github.com/signalfx/splunk-otel-collector",
-	"sumo":     "https://github.com/SumoLogic/sumologic-otel-collector",
-	"redhat":   "https://github.com/os-observability/redhat-opentelemetry-collector",
+	"core":          "https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol",
+	"contrib":       "https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib",
+	"aws":           "https://github.com/aws-observability/aws-otel-collector",
+	"grafana-agent": "https://github.com/grafana/agent",
```
Comment (dmitryax) [inline cmd/mdatagen/statusdata.go:18]: Is there another distribution from Grafana? If not, maybe just `grafana` given that we keep other distro names short?

Comment (mattdurham) [reply cmd/mdatagen/statusdata.go:18]: IMO This makes sense given what you are doing with the others. Normally we say grafana agent due to how common the term agent is.

**Comment by jpkrohling** [comment]
> Is there a good place to slot in a link to the documentation? 

Not in the contrib repository, I do remember seeing a list of distributions somewhere. I'll try to find it and send a PR there as well.

>  If not, maybe just grafana given that we keep other distro names short?

True, I'll change that

**Comment by jpkrohling** [comment]
PR is ready for review again.

File: exporter/lokiexporter/README.md [hunk 2]
```diff
@@ -4,11 +4,12 @@
 | Status        |           |
 | ------------- |-----------|
 | Stability     | [beta]: logs   |
-| Distributions | [contrib], [observiq] |
+| Distributions | [contrib], [grafana], [observiq] |
```
Comment (codeboten) [inline exporter/lokiexporter/README.md:7]: i couldn't find an import of the lokiexporter in the grafana-agent repo, is it there just not with the exporter module?

File: receiver/lokireceiver/README.md [hunk 3]
```diff
@@ -4,11 +4,12 @@
 | Status        |           |
 | ------------- |-----------|
 | Stability     | [alpha]: logs   |
-| Distributions | [contrib], [sumo] |
+| Distributions | [contrib], [grafana], [sumo] |
```
Comment (codeboten) [inline receiver/lokireceiver/README.md:7]: Same question as about the exporter

**Review by codeboten — COMMENTED** [review]
Looks like the span metrics processor was missed in this update

File: exporter/lokiexporter/README.md (hunk 2 above)
Comment (jpkrohling) [reply exporter/lokiexporter/README.md:7]: Great catch, you are absolutely right! The documentation page says:

> NOTE: otelcol.exporter.loki is a custom component unrelated to the lokiexporter from the OpenTelemetry Collector.

Source: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.loki/

File: receiver/lokireceiver/README.md (hunk 3 above)
Comment (jpkrohling) [reply receiver/lokireceiver/README.md:7]: Removed, for the same reason :) Great catch!

Comment (codeboten) [reply receiver/lokireceiver/README.md:7]: Was it removed and the change not pushed?

**Comment by jpkrohling** [comment]
For this PR, I'm including only the Flow components from the following page, as they are 1-1 to the OTel components. I'll open another PR if we think that the ones we use in static mode should be included as well.

I removed Loki receiver/exporter for now, as you are correct that they are not 1-1 to the ones we have in this repo here.

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**